contents/python/introduction-to-python.md

```diff
@@ -12,6 +12,15 @@ For absolute beginners, check out [this video](https://www.youtube.com/watch?v=N
 
 Both newbies and experienced programmers can also benefit from [The Python Tutorial](https://docs.python.org/3/tutorial/index.html), which aims to introduce readers to Python's unique features and style.
 
+## Virtual Environment
+
+When starting new projects or hopping on existing Python repositories, you will be recommended to install dependencies using [virtual environment](https://docs.python.org/3/tutorial/venv.html) to remove the possibility of dependencies conflict. This is a good practice especially when managing dependencies from different projects which relies on different python versions and packages.
```

Comment:
- `hopping on` -> `hopping into`
- `you will be recommended` -> `you are recommended`
- `using virtual environment` -> `using a virtual environment`
- `remove the possibility of dependencies conflict` -> `avoid dependency conflicts`
- `which relies` -> `which may rely`
- `python` -> `Python` (consistent with the official casing, and the rest of this document)